src/tropicalnumber.jl

```diff
@@ -0,0 +1,65 @@
+const pm_TropicalNumber_suppAddition = Union{pm_Min, pm_Max}
+const pm_TropicalNumber_suppScalar = Union{pm_Rational}
+
+#convert input to supported Scalar type
+pm_TropicalNumber{A,S}(scalar::Real) where {A <: pm_TropicalNumber_suppAddition, S <: pm_TropicalNumber_suppScalar} = pm_TropicalNumber{A,S}(convert(S,scalar))
+
+#pm_Rational es default Scalar
+pm_TropicalNumber{A}(scalar::Real) where A = pm_TropicalNumber{A,pm_Rational}(scalar)
+pm_TropicalNumber{A}() where A = pm_TropicalNumber{A,pm_Rational}()
+
+function Base.:(==)(x::pm_TropicalNumber{A,S},y::Real) where {A, S}
+    return x == pm_TropicalNumber{A,S}(y)
+end
+
+Base.:(==)(x::Real,y::pm_TropicalNumber{A,S}) where {A, S} = (y == x)
+
+#catch wrong typings for basic operations
+Base.promote_rule(::Type{<:pm_TropicalNumber{A, S1}}, ::Type{<:pm_TropicalNumber{A, S2}}) where {A, S1, S2} = pm_TropicalNumber{A, promote_type(S1,S2)}
+
+for op in (:+, :*, ://, :<)
+    @eval begin
+        Base.$(op)(x::pm_TropicalNumber{A, S}, y::pm_TropicalNumber{A, T}) where {A,S,T} = $(op)(promote(x,y)...)
+        Base.$(op)(x::pm_TropicalNumber{A}, y::pm_TropicalNumber{B}) where {A,B} =
+            throw(DomainError((x,y), "The operation $(string($op)) for tropical numbers with $A and $B is not defined"))
+    end
+end
+
+#at the moment we do not distinct between // and /, so / just refers to //
+Base.:/(x::pm_TropicalNumber{A1,S}, y::pm_TropicalNumber{A2, S}) where{A1, A2, S} = x//y
+
+#zero/one
+
+Base.zero(::Type{<:pm_TropicalNumber{A}}) where A = zero(pm_TropicalNumber{A}())
+Base.one(::Type{<:pm_TropicalNumber{A}}) where A = one(pm_TropicalNumber{A}())
+dual_zero(::Type{<:pm_TropicalNumber{A}}) where A = dual_zero(pm_TropicalNumber{A}())
+orientation(::pm_TropicalNumber{pm_Min}) = pm_Integer(1)
+orientation(::Type{<:pm_TropicalNumber{pm_Min}}) = pm_Integer(1)
+orientation(::pm_TropicalNumber{pm_Max}) = pm_Integer(-1)
+orientation(::Type{<:pm_TropicalNumber{pm_Max}}) = pm_Integer(-1)
+
+function pm_TropicalNumber{pm_Min, pm_Rational}(x::Float64)
+    if isinf(x)
+        if x > 0
+            return zero(pm_TropicalNumber{pm_Min}())
+        else
+            return dual_zero(pm_TropicalNumber{pm_Min})
+        end
+    else
+        return pm_TropicalNumber{pm_Min}(pm_Rational(x))
+    end
+end
```

Comment:
Is this correct that we do this to support `pm_TropicalNumber{pm_Min}(Inf)`?

Reply:
indeed it seems that we do, but `polymake` doesn't complain, so maybe we can leave it as it is?

Looking at the function itself I have two more comments @alexej-jordan 
* this function should return type-stable zeros, without relying on the fact that we allow only `pm_Rational`s as scalars, i.e.
```
function pm_TropicalNumber{pm_Min, S}(x::Float64) where S
    if isinf(x)
        if x > 0
            return zero(pm_TropicalNumber{pm_Min, S}())
        else
            return dual_zero(pm_TropicalNumber{pm_Min, S})
        end
    else
        return pm_TropicalNumber{pm_Min}(S(x))
    end
end
```
* `zero` gets an instance, while `dual_zero` just the type of `pm_TropicalNumber`. why is this so? Even if two calls are equivalent, in library code I'd prefer to see only one of them.

Reply:
@saschatimme yes this is correct. polymake itself also allows the construction of a tropicalnumber by passing the string `"inf` or `"-inf"`. as julia has the convenient Float64 values `Inf` and `-Inf`, these seem like a better possibility for our purposes

@kalmarek adjusted it accordingly. the difference you mentioned was most probably a relic of the construction.
also, i made it a bit more compact by distincting between pm_Min and pm_Max via orientation, so we do not have two different functions